project/SparkBuild.scala

```diff
@@ -435,7 +438,19 @@ object SparkBuild extends PomBuild {
       else x.settings(Seq[Setting[_]](): _*)
     } ++ Seq[Project](OldDeps.project)
   }
+}
 
+object Core {
+  lazy val settings = Seq(
+    unmanagedResourceDirectories in Compile += baseDirectory.value / "target" / "extra-resources",
+    resourceGenerators in Compile += Def.task {
+      val buildScript = baseDirectory.value + "/../build/spark-build-info"
+      val targetDir = baseDirectory.value + "/target/extra-resources/"
+      val command =  buildScript + " " + targetDir + " " + version.value
+      Process(Seq("/bin/bash", "-c", command)).!!
```

Comment:
Is the extra unmanagedResourceDirectory required? Alternatively, the resource generator could output the extra properties file into the `resourceManaged` directory (defaults to "resource_managed"), that way it would also automatically be picked up in the classpath.

Btw, I think that generating the file into `resourceManaged` would solve the problem with clean you reported on the mailing list.